qemu/tests/pvpanic.py

```diff
@@ -39,24 +38,26 @@ def setup_test_environment(test, params, vm, session):
         vm.reboot(session, timeout=timeout)
 
 
-def check_qmp_events(vm, event_name, timeout=360):
+def check_qmp_events(vm, event_names, timeout=360):
     """
     Check whether certain qmp event appeared in vm.
 
     :param vm: target virtual machine
-    :param event_name: target event name, such as 'GUEST_PANICKED'
+    :param event_names: a list of target event names,
+        such as 'GUEST_PANICKED'
     :param timeout: check time
     """
-    end_time = time.time() + timeout
+    def _do_check(vm, event_names):
+        for name in event_names:
+            if vm.monitor.get_event(name):
+                logging.info("Receive qmp %s event notification", name)
+                vm.monitor.clear_event(name)
+                return True
```

Comment:
Seems not correct if there is more than one event to be checked, I mean `True` should returned only if all the given events were received rather than just one of them.

Reply:
We can only get one of them after guest crashed. In other words, when no "events" param is set, "getting one of them" is right .

Reply:
We can only get one of them after guest crashed. In other words, when no "events" param is set, "getting one of them" is right .